src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs

```diff
@@ -223,7 +225,7 @@ internal static KernelCommandInvocation DoNugetRestore()
                     var resultMessage = new InstallPackagesMessage(
                             kernel.RestoreSources.OrderBy(s => s).ToArray(),
                             kernel.ResolvedPackageReferences.Select(s => $"{s.PackageName}, {s.PackageVersion}").OrderBy(s => s).ToList(),
-                            kernel.ResolvedPackageReferences.Where(currentPackageReferences.Add).Select(s => $"{s.PackageName}, {s.PackageVersion}").OrderBy(s => s).ToList(),
+                            kernel.ResolvedPackageReferences.Where(r => requested.Contains(r.PackageName, StringComparer.OrdinalIgnoreCase)).Where(currentPackageReferences.Add).Select(s => $"{s.PackageName}, {s.PackageVersion}").OrderBy(s => s).ToList(),
```

Comment:
Why `Contains` instead of `Equals`? This looks vulnerable to false matches, e.g. `Plotly.NET.Interactive` contains `Plotly.NET`.

Reply:
I want to add some logging too.

Reply:
When I wrote it I think I was worried that the names contained version.  It turns out they are just PackageNames so equals will be fine.

Reply:
Brain-fart
--- requested is a list of PackageNames so, it succeeds when the list contains the exact match modulo StringComparison.OrdinalIgnoreCase.

````
 var requested =
          kernel.RequestedPackageReferences
                .Except(kernel.ResolvedPackageReferences, PackageReferenceComparer.Instance)
                .Select(s => s.PackageName).OrderBy(s => s).ToList();
````

Reply:
I will add a test case using:
- Plotly.NET.Interactive contains Plotly.NET.